### PR TITLE
Create locator entities without association to other entities [11705]

### PR DESCRIPTION
**Pull request by SerranoePros**
-Modified some tests: now the creation and link of a locator is different.
-Database::insert_ddsendpoint_to_locator methods removed, they are no longer needed.

### CODE REVIEW

**Comment by JLBuenoLopez-eProsima** [comment]
This PR needs to be rebased

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

File: src/cpp/database/database.hpp [hunk 1]
```diff
@@ -436,19 +436,6 @@ class Database
...
-            // Add reader's locators to locators_by_participant_
-            locators_by_participant_[endpoint->participant->id][locator_it.first] = locator_it.second;
-            // Add reader's participant to participants_by_locator_
-            participants_by_locator_[locator_it.first][endpoint->participant->id] = endpoint->participant;
-            // Add endpoint to locator's collection
-            insert_ddsendpoint_to_locator(endpoint, locator_it.second);
-        }
-
         /* Add endpoint to topics's collection */
         endpoint->data.clear();
```
[the remark this answers is not in the document]
Comment (SerranoePros) [reply src/cpp/database/database.hpp:440]: Also I modified to insert first in endpoint's collection and then on the other maps, I think it makes sense

**Review by JLBuenoLopez-eProsima — DISMISSED** [review]
LGTM!

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM!

Rebased to `main` to ensure new changes does not conflict with changes introduced later on `main`. There are no new changes introduced in the rebase.

**Pull request merged**